src/overlays/actors/ovl_En_Dodongo/z_en_dodongo.c

```diff
@@ -21,59 +58,803 @@ const ActorInit En_Dodongo_InitVars = {
+        this->actor.shape.rot.y = this->actor.posRot.rot.y += this->tailSwipeSpeed;
+        tailPos.x = this->sphElements[1].dim.worldSphere.center.x;
+        tailPos.y = this->sphElements[1].dim.worldSphere.center.y;
+        tailPos.z = this->sphElements[1].dim.worldSphere.center.z;
+        func_80033260(globalCtx, &this->actor, &tailPos, 5.0f, 2, 2.0f, 100, 15, 0);
+        tailPos.x = this->sphElements[2].dim.worldSphere.center.x;
+        tailPos.y = this->sphElements[2].dim.worldSphere.center.y;
+        tailPos.z = this->sphElements[2].dim.worldSphere.center.z;
+        func_80033260(globalCtx, &this->actor, &tailPos, 5.0f, 2, 2.0f, 100, 15, 0);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dodongo/func_809F89CC.s")
+        if (this->colliderBody.base.atFlags & 2) {
+            Player* player = PLAYER;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dodongo/func_809F8A34.s")
+            if (this->colliderBody.base.at == &player->actor) {
+                Audio_PlayActorSound2(&player->actor, 0x83E);
+            }
+        }
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->colliderBody.base);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dodongo/func_809F8AD8.s")
+void EnDodongo_SetupDeath(EnDodongo* this, GlobalContext* globalCtx) {
+    SkelAnime_ChangeAnimTransitionStop(&this->skelAnime, &D_060013C4, -8.0f);
+    this->timer = 0;
+    Audio_PlayActorSound2(&this->actor, 0x3804);
+    this->actionState = DODONGO_DEATH;
+    this->actor.flags &= ~1;
+    this->actor.speedXZ = 0.0f;
+    EnDodongo_SetupAction(this, EnDodongo_Death);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dodongo/func_809F8B2C.s")
+void EnDodongo_Death(EnDodongo* this, GlobalContext* globalCtx) {
+    EnBom* bomb;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dodongo/func_809F8B7C.s")
+    if (this->skelAnime.animCurrentFrame < 35.0f) {
+        if (this->actor.params == EN_DODONGO_SMOKE_DEATH) {
+            EnDodongo_SpawnBombSmoke(this, globalCtx);
+        }
+    } else if (this->actor.dmgEffectTimer == 0) {
+        func_8003426C(&this->actor, 0x4000, 0x78, 0, 4);
+    }
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime) != 0) {
+        if (this->timer == 0) {
+            bomb = (EnBom*)Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_BOM, this->actor.posRot.pos.x,
+                                       this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0, 0, 6, BOMB_BODY);
+            if (bomb != NULL) {
+                bomb->timer = 0;
+                this->timer = 8;
+            }
+        }
+    } else if ((s32)this->skelAnime.animCurrentFrame == 0x34) {
+        Audio_PlayActorSound2(&this->actor, 0x387A);
+    }
+    if (this->timer != 0) {
+        this->timer--;
+        if (this->timer == 0) {
+            Item_DropCollectibleRandom(globalCtx, &this->actor, &this->actor.posRot.pos, 0x40);
```

Comment:
i think there might be an enum for the last arg

Reply:
Not for the random ones as far as I know.

Reply:
ah true